core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -276,7 +281,20 @@ abstract class BaseConsumerTest extends IntegrationTestHarness with Logging {
   protected def sendRecords(numRecords: Int, tp: TopicPartition) {
     (0 until numRecords).map { i =>
       this.producers(0).send(new ProducerRecord(tp.topic(), tp.partition(), s"key $i".getBytes, s"value $i".getBytes))
-    }.foreach(_.get)
+    }
+    this.producers(0).flush()
+  }
+
+  protected def sendCompressedRecords(numRecords: Int, tp: TopicPartition) {
+    val producerProps = new Properties()
+    producerProps.setProperty(ProducerConfig.COMPRESSION_TYPE_CONFIG, "gzip")
+    producerProps.setProperty(ProducerConfig.LINGER_MS_CONFIG, Long.MaxValue.toString)
+    val producer = TestUtils.createNewProducer(brokerList, securityProtocol = securityProtocol, trustStoreFile = trustStoreFile,
+      retries = 0, lingerMs = Long.MaxValue, props = Some(producerProps))
+    (0 until numRecords).foreach { i =>
+      producer.send(new ProducerRecord(tp.topic(), tp.partition(), s"key $i".getBytes, s"value $i".getBytes))
+    }
```

Comment:
Somewhat of a nitpick too, but we could extract a method for the following since it's duplicated in both `send*Records` methods.

```scala
    (0 until numRecords).foreach { i =>
       producer.send(new ProducerRecord(tp.topic(), tp.partition(), s"key $i".getBytes, s"value $i".getBytes))
     }
```